src/training_hub/cli.py

```
@@ -0,0 +1,638 @@
+
+    # Step 3: Check for missing required params
+    missing = []
+    for flag, spec in param_defs.items():
+        if spec.get("required"):
+            python_name = _flag_to_python_name(flag)
+            if python_name not in config_values or config_values[python_name] is None:
+                missing.append(flag)
+
+    if missing:
+        print(f"error: the following arguments are required: {', '.join(missing)}", file=sys.stderr)
+        print(f"hint: provide them via CLI flags or in a YAML config file with --config", file=sys.stderr)
```

Ruff (0.15.20)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)